### DISCUSSION TITLE
ZIM Task to Calendar

### DISCUSSION THREAD

**Discussion by theoky** [zim-desktop-wiki/zim-desktop-wiki#2654] [Show and tell]
I've been using ZIM for a long time to take notes and manage my tasks. As I have a significant amount of meetings, finishing all important tasks has become challenging. As there is some sort of meeting creep, tasks get delayed additionally.

So I tried to solve that for me in a way where my tasks (managed in ZIM) become appointments in my Outlook calender as well (which I have seen already somewhen around 2006 done by a tool and long before skedpal and the like). As this is not managable manually, I wrote a small standalone Python tool (no plugin) as a hobby project which does basically the following:

* user: 
	* enriches tasks in ZIM with meta information like duration, e.g. a task with the text "do something ~30m~" will take 30min (probably).
* Tool:
	* build an internal scheduling calendar with my available working time
	* get all appointments from my calendar and remove them from the available working time, thus creating a digital twin calendar.
	* add some default appointments (e.g. lunch break) - or not.
	* - - now the internal calendar contains only time left which is available for working on tasks.
	* schedule all tasks with a given duration as appointments. Tasks with no duration specified will take a default amount of time (e.g. 6min). Tasks longer than an hour will be split into subtasks of max. one hour length.
	* schedule according to priority and tags (e.g. ILM (Ivy Lee Method) tasks should be handled first, because they should be done today, then legal task, and so on)
	* see what can't get scheduled and drops out.

The resulting schedule can then be presented as a ZIM page or additionally as appointments in Outlook, so that they block time for work instead for meetings. 
As a bonus, it works without cloud.

This tool is currently only for me - and thus also only usable for me.

If there is some interest, I would spend some time to prepare a (slow) transition from a hobby project to a public repo (the usual stuff, documentation, more tests, etc.), so that others can use it - or adapt it - as well.

Would there be interest in such a tool?

**Comment by introt** [comment]
> or additionally as appointments in Outlook, so that they block time for work instead for meetings.
> As a bonus, it works without cloud.

I'd be interested to see/hear how your tool works - I've cobbled together something similar (on top of the corpse of the ill-fated [zim2anki](https://github.com/introt/zim2anki)) to turn tasks with deadlines (eg. `< 2022-02-02`) into an ics file/iCal feed, but that's a one-way conversion and requires some cloud to function. Now that I've mentioned it, I'd advice you to learn from my mistakes: publish whatever you can as-is, perhaps with a warning that it's being actively worked on, and then add the polish iteratively - otherwise you risk never getting anything out as the bar to make the first push keeps rising with time.

**Reply by brucewestfall** [reply]
If Linus Torvalds did not publish early, the internet and entire ecosystem of computers would be different.  And much sadder.

**Comment by theoky** [comment]
As I don't want do contribute to sadness 😄 I've published what I have here: [LurchCal](https://github.com/theoky/lurchcal).
It is still in a very early stage and raw, so don't expect too much...
